src/DIRAC/WorkloadManagementSystem/Executor/InputData.py

```diff
@@ -379,8 +385,6 @@ def __getSiteCandidates(self, okReplicas, vo):
                     if seStatus["DiskSE"]:
                         # Sets contain only unique elements, no need to check if it's there
                         diskLFNs.add(lfn)
-                        if lfn in tapeLFNs:
-                            tapeLFNs.remove(lfn)
```

Comment:
Why remove this? What error does this fix?

Reply:
There's an explanation in the PR description. It also makes this test https://github.com/DIRACGrid/DIRAC/pull/6647/files#diff-a9efd82e4d66034f42ef8bf065f98b424b0f8ce9b0b9e8c58b6ebe9fccf64232R152-R165 to pass.

Reply:
Ok, Sorry I missed that!
I don't understand the logic of this function, or how the output is used to be honest.